index.js

```diff
@@ -7,24 +7,39 @@ const fs = require('fs');
 
 async function run() {
   try {
-    const msys2base = await tc.downloadTool('http://repo.msys2.org/distrib/x86_64/msys2-base-x86_64-20190524.tar.xz');
-    const tar = await io.which('tar', true);
+    if (process.platform != 'win32') {
+      core.setFailed("MSYS2 does not work on non-windows platforms; please check the 'runs-on' field of the job");
+    }
 
-    const dest = path.join(process.env['RUNNER_TEMP'], 'msys')
+    const tar = await io.which('tar', true);
 
-    // For some reason, GNU Tar on Windows expects paths to be slash-separated
-    const normalizedDest = dest.replace(/\\/g, '/')
+    const tmp_dir = process.env['RUNNER_TEMP']
+    if (!tmp_dir) {
+      core.setFailed('environment variable RUNNER_TEMP is undefined');
+      return
+    }
+    const dest = path.join(tmp_dir, 'msys')
 
     await io.mkdirP(dest)
 
-    await exec.exec(`"${tar}"`, ['-x', '-J', '--force-local', '-C', normalizedDest, '-f', msys2base])
+    await exec.exec(`"${tar}"`, [
+      '-x', '-J', '--force-local',
+      // For some reason, GNU Tar on Windows expects paths to be slash-separated
+      '-C', dest.replace(/\\/g, '/'),
+      '-f', await tc.downloadTool('http://repo.msys2.org/distrib/x86_64/msys2-base-x86_64-20190524.tar.xz')
+    ])
 
-    fs.writeFileSync(
-      path.join(dest, 'msys2do.cmd'),
-      fs.readFileSync(path.join(__dirname, 'msys2do.cmd'))
-    )
+    let cmd = path.join(dest, 'msys2do.cmd')
+    fs.writeFileSync(cmd, ['setlocal', 'set MSYS2_PATH_TYPE=strict', `%~dp0\\msys64\\usr\\bin\\bash.exe -lc "cd $OLDPWD && %*"`].join('\n\r'))
```

Comment:
This used to be interactive (`-i`). I don't remember exactly why, but I think I needed that.

Reply:
I removed it because I didn't find it to be required in actions/starter-workflows#95, and due to actions/starter-workflows#114. Nevertheless, I added it now.